src/error.rs

```diff
@@ -23,6 +23,9 @@ pub enum Error {
     #[cfg(feature = "std")]
     /// A custom Scroll error for reporting messages to clients
     Custom(String),
+    #[cfg(not(feature = "std"))]
+    /// A custom static Scroll error for reporting messages to clients
+    Custom(&'static str),
```

Comment:
I'd like to understand why this change occurred; ideally fixing the tests to run on no std shouldn't require changing the error type here.

If we are required to fix this, I would prefer to name this variant Static and have it always present, if possible; ideally the std flag is additive and doesn't change the meaning of the enum variant when active/not active. Alternatively we could explore Cow and alloc feature on core, but probably a static variant is best

Reply:
@m4b neither the `Custom` nor `Static` errors are required - they are both used for user errors from what i can tell, but in the nostd env, it would not be possible to have a custom error for users - so adding a static one is the simplest way to solve it.  I think `CustomStatic` is a clearer error type - what do you think?

Reply:
that seems ok, yea; honestly i would prefer Cow (for this exact) usecase but it only exists in alloc, so we'd need an alloc flag, and then in any baremetal system you'd still have the same problem, so CustomStatic seems fine I guess

Reply:
thx, so any changes needed for this one still, or is it good to merge?